src/context.rs

```diff
@@ -17,6 +22,13 @@ impl Context for StaticContext {
             .map(|v| v.to_string())
             .ok_or_else(|| failure::err_msg(format!("key not found: {}", key)))
     }
+
+    fn exec(&self, processor: &str, query: &str) -> Result<String, failure::Error> {
+            match processor {
+                "upper" => Ok(query.to_uppercase()),
+                _ => unimplemented!("implement it in your context")
+            }
+        }
```

Comment:
Nopes, this is a dummy implementation only for testing. StaticContext is a public api, can't be used like this.

Reply:
If it is really needed (I thought implementing a function in trait means I do not have to implement it unless I am overwriting), put just the `unimplemented!()` statement as body.